File: 0932-monotonic-array/0932-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        int n = nums.size();
        int inc = 0, dec = 0;

        for(int i = 0; i < n - 1; i++) {
            if(nums[i] <= nums[i+1]) inc++;
            if(nums[i] >= nums[i+1]) dec++;
        }

        if (inc == nums.size() - 1 || dec == nums.size() - 1)
            return true;
            
        return false;  
    }
};

File: 0932-monotonic-array/0932-monotonic-array.go
func isMonotonic(nums []int) bool {
    n := len(nums)

    inc, dec := 0, 0

    for i := 0; i < n-1; i++ {
        if nums[i] <= nums[i+1] {
            inc += 1
        }
        if nums[i] >= nums[i+1] {
            dec += 1
        }
    }

    if (inc == n-1) || (dec == n-1) {
        return true
    }

    return false
}